engine.py

```diff
@@ -499,9 +498,24 @@ def _create_dialog(self, title, bundle, widget, parent):
         :param parent: The parent QWidget for the dialog
         """
 
+        from tank.platform.qt import QtCore
+
         # call the base implementation to create the dialog:
         dialog = tank.platform.Engine._create_dialog(self, title, bundle, widget, parent)
 
+        if dialog.parent():
+            # parenting crushes the dialog's style. This seems to work to reset
+            # the style to the dark look and feel in preparation for the
+            # re-application below.
+            dialog.parent().setStyleSheet("")
+        else:
+            # no parent, so style should be ok. set window flag to be on top
```

Comment:
Since `_get_dialog_parent()` now returns the main window, when does this case happen? I think a brief comment to make it easier to understand why we have these two cases would be useful for code readers